manager/volume.go

```diff
@@ -305,8 +302,12 @@ func (m *VolumeManager) Attach(name, nodeID string, disableFrontend bool, attach
 	if v.Status.State != types.VolumeStateDetached {
 		return nil, fmt.Errorf("invalid state to attach %v: %v", name, v.Status.State)
 	}
-	if err := m.CheckEngineImageReadiness(v.Spec.EngineImage); err != nil {
-		return nil, errors.Wrapf(err, "cannot attach volume %v with image %v", v.Name, v.Spec.EngineImage)
+
+	if isReady, message, err := m.CheckEngineImageReadinessForVolume(v.Spec.EngineImage, v.Name, nodeID); !isReady {
+		if err != nil {
```

Comment:
The manager doesn't have a logger. Also, this function return value is a user-facing value so handle the err with `if-else`

Reply:
I see that makes sense, you are returning this to the api caller, good catch.
In regard to the logger you can just do.
logrus.withError() but it's unecessary here since we area returning it to the api caller.